opacclient/opacapp/src/main/java/de/geeksfactory/opacclient/frontend/StarredFragment.java

```diff
@@ -256,6 +290,166 @@ protected void export() {
                 getResources().getString(R.string.share)));
     }
 
+    public void exportToStorage() {
+        Intent intent = null;
+        //android 4.4+; use Storage Access Framework
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+            intent = new Intent(Intent.ACTION_CREATE_DOCUMENT);
+            intent.addCategory(Intent.CATEGORY_OPENABLE);
+            // Create a file with the requested MIME type.
+            intent.setType("application/json");
+            intent.putExtra(Intent.EXTRA_TITLE, "share.json");
+            startActivityForResult(intent, REQUEST_CODE_EXPORT);
+        } else {        // <android 4.4; share raw json file
+            if(Environment.getExternalStorageState()==Environment.MEDIA_MOUNTED) {
+                File dest = new File(getActivity().getExternalCacheDir(),"starred_export.json");
```

Comment:
We could, but then the onus of actually creating/storing the JSON file would now be on the owner since with `Intent.EXTRA_TEXT`, we would have to pass the raw JSON string, which seems pointless from a user's POV. For example, it would be easy for a user to accidentally edit this content before he saves it somewhere. Which is why I believe we need the cached file.